Adds the merge_patient function

fredkingham opened the pull request:
Adds in intrahospital_api.merge_patient.merge_patient that takes two patients and merges one onto the other.

Commit:
Adds the merge_patient function

Adds in intrahospital_api.merge_patient.merge_patient that takes two
patients and merges one onto the other.

davidmiller commented on intrahospital_api/merge_patient.py:
```diff
@@ -0,0 +1,146 @@
+
+
+@transaction.atomic
+def merge_patient(old_patient, new_patient):
```
Suspect t's clearer in the calling code if these are named kwargs

davidmiller commented on intrahospital_api/merge_patient.py:
```diff
@@ -0,0 +1,146 @@
+    new_mrn = new_patient.demographics().hospital_number
+
+    for patient_subrecord in subrecords.patient_subrecords():
+        if not issubclass(patient_subrecord, models.PreviousMRN):
```
What does this exclude ? Do we definitely catch everything ? OTTOMH is ICUhandover in `load_patient` ?

Commit:
Change the merge patient arguments to be keyword arguments

As this is a function we do not want people to misread, make sure that
we force keyword arguments so that the caller has to know what they are
doing.

Commit:
Adds a special test for blood culture isolates in the merge_patient tests

Adds a blood culture test just for merge patients as its one of the only
models with foreign keys.

Commit:
Adds to the merge patient the copying of other models that aren't subrecords

Copy accross models that aren't patient subrecords and make sure that
the status models are set correctly

Commit:
Make sure that we copy over tagging values when we merge patients

Commit:
Adds a unit test for merge patient to check we copy over observations

Commit:
Adds comments and doc strings to merge_patient

Commit:
Removes the unused new_mrn argument in some of merge_patient functions

Commit:
Fixes a comment in merge patient to be more readable

Commit:
Moves the is_episode_subrecord logic into update_singleton/copy_non_singletons

Commit:
Reformats merge_patient with black

Commit:
Adds hardcoded include/ignore lists for models in merge_patient

Hard codes a list of patient/episode related models that should be
copied over. Adds an patient/episode ignore lists in the test.

If a related model is not in either of these lists an error is raised by
the unit test. This forces us to explicitly make sure that related
models are considered when they are added.

Commit:
Fixes up function names/docstrings in merge_patient and test_merge_patient

Hopefully this makes it more intuitively readable.

Commit:
Removes now unused functions from merge_patient

Commit:
Renames move_tagging to update_tagging as it does not actually move tags

In merge_patient, rename move_tagging to update_tagging
